### PR TITLE
Fix issue 378: duplicate imports

### PR DESCRIPTION
**Pull request by shibumi**
We had a few unnecessary duplicate imports.
This commit removes the additional "from-import" lines,
and just uses the full path for the import.

Please fill in the fields below to submit a pull request.  The more information
that is provided, the better.

**Fixes issue #**:

Fixes issue #378 

**Description of the changes being introduced by the pull request**:

Use imports and not `from .. import ..`-lines. I also realized that the latter is against our Code Style Guidelines.. ^^

**Please verify and check that the pull request fulfills the following
requirements**:

- [x] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [x] Tests have been added for the bug fix or new feature
- [x] Docs have been added for the bug fix or new feature

### CODE REVIEW

**Comment by shibumi** [comment]
I have tested the PR with calling `python tests/runtests.py`. I hope this is enough.

**Review by trishankatdatadog — APPROVED** [review]

**Comment by shibumi** [comment]
ah okay, the linting fails, because the line is too long.

**Comment by shibumi** [comment]
Mh is not that easy to satisfy the linter.. it's complaining about the variable name `m` or `ml`, because it has to be snakecase..

**Comment by shibumi** [comment]
Mh I have added a pylint exception now for the few lines. Otherwise I couldn't make this PR work, I have no idea if it's worth it to fix a few double imports.

**Review by lukpueh — COMMENTED** [review]
Thanks for fixing the *lgtm* warning, @shibumi!

This might be a matter of taste, but a local scope module shortcut does not make a lot of sense to me, especially if then used only once.

I suggest do either make the shortcut module scope, which is what we had before this PR, or use the full module path in the call statement and wrap the line appropriately, which makes it look more like java but is what we usually do in in-toto/tuf/securesystemslib code.

I also like what [google's style guide suggests](https://google.github.io/styleguide/pyguide.html#22-imports), i.e. to:
```python
from full.path.to.package import module
...
module.function()
```

The caveat I see there, is that the module name by itself should be meaningful, which is not really the case for `interface`, and must not conflict with any locally defined variables, which is not the case for `link`. But we could work around this. :)

What do you think?

**Comment by shibumi** [comment]
Hi @lukpueh 
> This might be a matter of taste, but a local scope module shortcut does not make a lot of sense to me, especially if then used only once.

Sounds good, I think I will try the Java-like style first to stick with our current styleguide.

**Comment by shibumi** [comment]
@lukpueh I have modified the lines, we are using linewraps now. However, my IDE pycharm states, that we are violating PEP8 now.

IIRC, PEP8 compliance has been discussed in the newest code style issue, too. Right?

File: in_toto/verifylib.py [hunk 1]
```diff
@@ -222,7 +221,10 @@ def run_all_inspections(layout):
 
     # Dump the inspection link file for auditing
     # Keep in mind that this pollutes the verifier's (client's) filesystem.
-    filename = FILENAME_FORMAT_SHORT.format(step_name=inspection.name)
+    # ignore snakecase for pylint to satisfy line limit
+    # pylint: disable=C0103
+    m = in_toto.models.link
+    filename = m.FILENAME_FORMAT_SHORT.format(step_name=inspection.name)
```
Comment (lukpueh) [inline in_toto/verifylib.py]: [ping](https://github.com/in-toto/in-toto/pull/379#pullrequestreview-459873487)

**Review by lukpueh — CHANGES_REQUESTED** [review]
Thanks for the quick patch, @shibumi. But this is not exactly what I meant. I think escaping newlines is really ugly and should only be done if the other alternatives don't work for some reasons.

The preferred (and [PEP8 compliant](https://www.python.org/dev/peps/pep-0008/#indentation)) way is to do hanging indent with an extra indentation level, or aligned indent.

I think in in-toto we almost exclusively use hanging over aligned indent, albeit with arguments on the first line, which is disallowed by PEP8 but IMO doesn't impede readability.

File: in_toto/verifylib.py (hunk 1 above)
Comment (shibumi) [reply in_toto/verifylib.py]: oops, looks like I missed one

**Comment by shibumi** [comment]
Ok, I tried using a hanging indent for most of them, however one line is still too long. I tried renaming the variable, but I think making `sublayout_link_dir` to `sub_link_dir` would destroying the readability of the variable name. It will be more difficult to tell, what the variable is doing, although we use the variable only in the next line. What do you think?

File: in_toto/util.py [hunk 2]
```diff
@@ -170,7 +168,8 @@ def import_public_keys_from_files_as_dict(filepaths, key_types=None):
   for idx, filepath in enumerate(filepaths):
 
     if key_types[idx] == KEY_TYPE_ED25519:
-      key = import_ed25519_publickey_from_file(filepath)
+      key = securesystemslib.interface.import_ed25519_publickey_from_file(
+        filepath)
```
Comment (lukpueh) [inline in_toto/util.py]:
```suggestion
          filepath)
```

File: in_toto/verifylib.py [hunk 3]
```diff
@@ -1311,11 +1311,11 @@ def verify_sublayouts(layout, chain_link_dict, superlayout_link_dir_path):
         # name relative the the current link directory path, i.e. if there
         # are multiple levels of sublayout nesting, the links are expected to
         # be nested accordingly
-        sublayout_link_dir = SUBLAYOUT_LINK_DIR_FORMAT.format(
-            name=step_name, keyid=keyid)
+        sub_link_dir = in_toto.models.layout.SUBLAYOUT_LINK_DIR_FORMAT.format(
```
Comment (shibumi) [inline in_toto/verifylib.py:1314]: This is what i mean.. I am not sure if `sub_link_dir` is okay as variable name... However we only use it in the next line. What do you think?

File: in_toto/util.py [hunk 4]
```diff
@@ -241,10 +240,12 @@ def prompt_import_ed25519_privatekey_from_file(filepath):
   """
   password = None
   try:
-    import_ed25519_privatekey_from_file(filepath, password)
+    securesystemslib.interface.import_ed25519_privatekey_from_file(
+      filepath, password)
```
Comment (lukpueh) [inline in_toto/util.py]:
```suggestion
        filepath, password)
```

File: in_toto/util.py [hunk 5]
```diff
@@ -241,10 +240,12 @@ def prompt_import_ed25519_privatekey_from_file(filepath):
...
   except securesystemslib.exceptions.CryptoError:
     password = prompt_password()
-  return import_ed25519_privatekey_from_file(filepath, password)
+  return securesystemslib.interface.import_ed25519_privatekey_from_file(
+    filepath, password)
```
Comment (lukpueh) [inline in_toto/util.py]:
```suggestion
      filepath, password)
```

File: in_toto/verifylib.py [hunk 6]
```diff
@@ -144,7 +142,8 @@ def load_links_for_layout(layout, link_dir_path):
       for keyid in [authorized_keyid] + list(layout.keys.get(authorized_keyid,
           {}).get("subkeys", {}).keys()):
 
-        filename = FILENAME_FORMAT.format(step_name=step.name, keyid=keyid)
+        filename = in_toto.models.link.FILENAME_FORMAT.format\
+          (step_name=step.name, keyid=keyid)
```
Comment (lukpueh) [inline in_toto/verifylib.py]:
```suggestion
        filename = in_toto.models.link.FILENAME_FORMAT.format(
            step_name=step.name, keyid=keyid)
```

File: in_toto/verifylib.py [hunk 7]
```diff
@@ -222,7 +221,8 @@ def run_all_inspections(layout):
 
     # Dump the inspection link file for auditing
     # Keep in mind that this pollutes the verifier's (client's) filesystem.
-    filename = FILENAME_FORMAT_SHORT.format(step_name=inspection.name)
+    filename = in_toto.models.link.FILENAME_FORMAT_SHORT.format(
+      step_name=inspection.name)
```
Comment (lukpueh) [inline in_toto/verifylib.py]:
```suggestion
        step_name=inspection.name)
```

File: in_toto/verifylib.py [hunk 8]
```diff
@@ -1311,11 +1311,11 @@ def verify_sublayouts(layout, chain_link_dict, superlayout_link_dir_path):
         # name relative the the current link directory path, i.e. if there
         # are multiple levels of sublayout nesting, the links are expected to
         # be nested accordingly
-        sublayout_link_dir = SUBLAYOUT_LINK_DIR_FORMAT.format(
-            name=step_name, keyid=keyid)
+        sub_link_dir = in_toto.models.layout.SUBLAYOUT_LINK_DIR_FORMAT.format(
+          name=step_name, keyid=keyid)
```
Comment (lukpueh) [inline in_toto/verifylib.py]:
```suggestion
            name=step_name, keyid=keyid)
```

File: in_toto/verifylib.py (hunk 3 above)
Comment (lukpueh) [inline in_toto/verifylib.py:1314]: 👍  works for me.

**Review by lukpueh — CHANGES_REQUESTED** [review]

**Comment by lukpueh** [comment]
Added **double** indentation for the hanging indent and replaced one escaped newline. Feel free to batch apply my suggestions and squash into your commit. I think we can merge then. :)

**Comment by shibumi** [comment]
@lukpueh please verify if this has worked correctly. I am not sure about my git skills here. What I did is:

1. applied all changes via github GUI
2. checked out this PR directly via one of my git alias'es
3. rebased the PR branch on my fork branch
4. force-pushed to this PR again.

**Review by lukpueh — APPROVED** [review]
LGTM

**Comment by trishankatdatadog** [comment]
Great job, guys! Sorry this was not as simple as expected...

**Comment by lukpueh** [comment]
No worries, @trishankatdatadog. Maybe we should put our impeccable lgtm rating to the README.md? I reckon there is a badge for it?

**Comment by trishankatdatadog** [comment]
I think let's try enabling CodeQL instead (appears to be the next evolution of LGTM now that it has been acquired by GitHub). Could you see if we can turn on the [public beta](https://github.blog/changelog/2020-05-06-github-advanced-security-code-scanning-now-available-in-limited-public-beta/)?

**Comment by shibumi** [comment]
In case you want a badge for LGTM.. here is one:

https://img.shields.io/lgtm/grade/python/github/secure-systems-lab/securesystemslib

(built via https:/shields.io )

**Pull request merged**